server/manager/src/main/java/org/apache/accumulo/manager/Manager.java

```diff
@@ -568,10 +573,13 @@ private class MigrationCleanupThread implements Runnable {
     @Override
     public void run() {
       while (stillManager()) {
-        if (!migrations.isEmpty()) {
+        // migrations are stored in the metadata tables which cannot be read until the
+        // TabletGroupWatchers are started
+        if (watchersStarted.get() && numMigrations() > 0) {
           try {
             cleanupOfflineMigrations();
             cleanupNonexistentMigrations(getContext());
+            cleanupDeletedMigrations();
```

Comment:
Previously, when TServers were deleted, the migrations to these TServers were deleted every call to `Manager.update()` which is called every 5 seconds (see `LiveTServerSet.scanServers()`). Now this is done every 5 minutes (see the `sleepUninterruptibly(CLEANUP_INTERVAL_MINUTES, MINUTES);` below). Not sure if the extra delay matters or not, so making note in case it does.